### ISSUE TITLE
Cannot read property of Slice on sort field reordering

### ISSUE THREAD

**Issue by jay-cohen** [directus/directus#11374]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

When reordering an existing collections nested items (Many to Many) in a parent "feed" collection the error {message: "Cannot read property 'slice' of undefined", extensions: {code: "INTERNAL_SERVER_ERROR"}} appears when saving. 

The reordering also doesn't work the first time resulting in a second attempt which nulls the nested title leaving the following placeholder "--:--"

### To Reproduce

Create a collection that has a nested Many to Many relationship with another collection and enable reordering on the nested collection. When attempting to save an existing collection after reordering the error message is shown or the reordering doesn't persist in the view layout (reverts back to original)

### Errors Shown

{message: "Cannot read property 'slice' of undefined", extensions: {code: "INTERNAL_SERVER_ERROR"}}

### What version of Directus are you using?

9.5.0

### What version of Node.js are you using?

v14.16.0

### What database are you using?

MySQL

### What browser are you using?

Chrome, Safari

### What operating system are you using?

MacOS

### How are you deploying Directus?

Locally

**Comment by jay-cohen** [comment]
Also, just noted when creating a new feed the UI doesn't work in that it repeats a field for every reorder. Example: If I add two cards to a feed and reorder one above the other I now have 4 cards showing. If I reorder again I now have 8 cards showing.

**Comment by licitdev** [comment]
Hi @jay-cohen, I am unable to reproduce your issue. Will it be possible to provide detailed steps or a screen recording to reproduce your issue?

I have enabled reordering on the main tables and nested tables.

https://user-images.githubusercontent.com/26413686/152296840-191b9642-09f7-415f-a63e-4425fdab89a5.mov

**Comment by jay-cohen** [comment]
@licitdev, Here's a video showcasing the issue.

https://user-images.githubusercontent.com/49128160/152319494-3f789e0e-f3f1-4ab8-9d40-748e6d3012e9.mov

**Comment by licitdev** [comment]
@jay-cohen Thank you for the screen recording. The issue lies with the sort field being `card`, which is a foreign key. You should create a separate integer field in `feeds_cards` to be used as the sort field. Please kindly revert whether the issue persists after you have implemented the change.

**Comment by jay-cohen** [comment]
@licitdev - I implemented the change above and it works perfectly. Thanks for your help.

**Comment by licitdev** [comment]
@rijkvanzanten Will it be good to push a PR that prevents the using primary / foreign keys as a sort field? I think it will prevent this issue from recurring.

**Comment by rijkvanzanten** [comment]
Yes absolutely. That also ties in with https://github.com/directus/directus/issues/11357
